### PR TITLE
Add cram support to Allelecounter module

### LINKED ISSUE #1012: [FEATURE] Add CRAM file support to allelecounter
## Is your feature request related to a problem? Please describe

Current allelecounter module works only with bam files. For use in sarek support for cram files is needed.

## Describe the solution you'd like

An extra parameter for the allecounter module with the genome fasta is needed to allow passing cram files.
When BAM files are used an empty list can be passed instead of the fasta file.

### PR DESCRIPTION
**Pull request by fbdtemme**
## PR checklist

Closes #1012

This PR adds support for CRAM files to the allelecounter module by adding an optional fasta input parameter

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by maxulysse** [comment]
Could you update the yaml file with the added fasta file?

**Comment by fbdtemme** [comment]
Forgot to add that file to the commit. Fixed now.

**Comment by maxulysse** [comment]
Can you change `bam` and `bai` to `input` and `input_index` as we already did with @FriederikeHanssen in strelka/manta modules ?
That would be perfect.

**Review by maxulysse — APPROVED** [review]
I can only now say that it's perfect
<3

File: modules/allelecounter/main.nf [hunk 1]
```diff
@@ -19,20 +19,24 @@ process ALLELECOUNTER {
...
 
     script:
     def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def reference_options = fasta ? "-r $fasta": ""
+
     """
     alleleCounter \\
         $options.args \\
         -l $loci \\
         -b $bam \\
```
Comment (maxulysse) [inline modules/allelecounter/main.nf:38]:
```suggestion
        -b $input \\
```

**Comment by maxulysse** [comment]
> I can only now say that it's perfect <3

Spoke too fast, we forgot a `bam` -> `input` in the process.
Hope you won't mind I committed that suggestion.

**Comment by fbdtemme** [comment]
I just saw it too and commited it just after you.

**Pull request merged**